functional_tests.go

```diff
@@ -85,14 +85,46 @@ func getFilePath(filename string) (filepath string) {
 	return
 }
 
+// Implements a type of reader which returns a
+// constant stream of random content
+type randomReader struct {
+	TotalBytes       int64
+	RandomSeedChar   rune
+	currentReadBytes int
+}
+
+func (r *randomReader) Read(p []byte) (n int, err error) {
```

Comment:
Can you add some comments in this function ? @harshavardhana, it is quite complicated to read